### ISSUE TITLE
"No passwords yet" issue (PostgreSQL only)

### ISSUE THREAD

**Issue by atsp** [fcturner/passwords#161]
Hello.
I have made fresh install of owncloud and Passwords to my server. The problem is - new passwords are stored in database (oc_passwords table) but are not shown in Owncloud's Passwords section.
Tried different local users, tried different browsers Safari/Firefox@Mac and Firefox@Win7 - no luck. I'm on latest versions of owncloud (9.0.0s) and Passwords (17.2).
PHP returns correct time/date, but in oc_passwords creation_date are set to 1970-01-01 for every item.
Can you help me?

`Error 	PHP 	Invalid argument supplied for foreach() at /var/www/html/owncloud/apps/passwords/service/passwordservice.php#27 	2016-03-26T17:55:27+00:00`

Ubuntu 14.04 LTS Server
PHP Version 5.5.9-1ubuntu4.14
Apache 2.4.7-1ubuntu4.9
postgresql  9.3+154ubuntu1

**Comment by fcturner** [comment]
Please download the GitHub master version and try again. 

A creation date of 1970 is perfectly normal. In previous versions this column held the actual creation date. But from v17 on, I blinded this for database admins and moved it to an encrypted value in the properties column. I wanted perfect privacy for users, even from their database administrator. So they are never able to even retrieve the date you changed a password.

**Comment by atsp** [comment]
Thank you for such a fast answer!
I've made double full reinstall for owncloud (with deleting owncloud folder, droping all tables in DB and clearing owncloud data folder - it's all, right?) with fresh git clone https://github.com/fcturner/passwords.git.
Now I'm getting these errors instantly after accessing Passwords:
`Error: Could not load categories.`
`Error: Could not load passwords.`

`Error 	PHP 	Undefined constant 'OCA\Passwords\Db' at /var/www/html/owncloud/apps/passwords/db/passwordmapper.php#2 	2016-03-26T19:06:52+00:00
Error 	PHP 	Undefined constant 'OCA\Passwords\Db' at /var/www/html/owncloud/apps/passwords/db/categorymapper.php#2 	2016-03-26T19:06:52+00:00 `
And after that it's infinitely "Decrypting passwords..."

(both Password tables are present and empty.)

**Comment by fcturner** [comment]
Hm, I thought I fixed that. I'll dive into it immediately. Thanks for making my app better!

**Comment by fcturner** [comment]
Can you try again? #162 might've solved it. 
Just reGit, it's only about the app files since the error is front side.

**Comment by atsp** [comment]
Yes, I confirm that error is gone. Thank you!
But still I cannot add new passwords. According to DB, they are added, but with empty 'loginname', 'address' and 'notes' fields in 'oc_passwords', and app cannot see them; 'userid', 'website' and encrypted data in 'pass' and 'properties' fields are filled.

**Comment by fcturner** [comment]
That's how it should be. All properties like login name, address, notes, categories, strength, length, are in the properties column. Hidden and encrypted. The other columns are there to support previous versions. But you should be able to view them in the app. 

Any browser inspector or log errors?

**Comment by atsp** [comment]
Got a freshly installed Win10 and latest Firefox. Same thing - no passwords..

![op](https://cloud.githubusercontent.com/assets/9769008/14064621/3eb1bb7a-f412-11e5-9431-d7c1b665f366.jpg)

**Comment by fcturner** [comment]
I did a fresh install with OC 9 and latest Passwords git clone too (17.3), works perfectly. 

It seems your security policy is blocking some functions. I tested on Windows 7 with Firefox and Chrome, OSX 10.11 with Chrome and Safari and on iPhone and iPad both with iOS 9.3 and mobile Safari. All works very well. 

I'm thinking of a server configuration at your side. Did you overwrite the security policy in config.php in /owncloud/config?
Does the ownCloud log tell you anything? Might be PostgreSQL related to. Horrible database to support for app developers, but I try my best :wink:

**Comment by atsp** [comment]
Finally, I located problem.
But before that I installed another one server (now based on Debian 8.3) and took different notebook with Ubuntu as a client. The problem was still there. Spent almost whole my weekend with it, and now my wife looks at me suspiciously :-D
I installed mariaDB and reinstalled owncloud to use it and VOILA - it instantly works as it should!

So at least fresh installation of Passwords app for Owncloud doesn't work with PgSQL :( Is it possible for you to take a look for this problem?

**Comment by fcturner** [comment]
I cannot find a solution for now. Version 18 will be released this or next week, introducing sharing. I dropped support for PostgreSQL for this version. I'll do my best (please help!) to support it in 18.1 again, but I don't think this will be solved in this or next week :cry:

**Comment by richpuch** [comment]
I have run into the same problem, I did a clean install of version OC8.1 and then upgrade to stable 8.2.3.  But when I try passwords I get the error:
Error: Could not load passwords
and the constant:
Decrypting passwords
I did a clean git clone https://github.com/fcturner/passwords.git  /home/www/owncloud/apps/passwords
with no luck.

**Comment by fcturner** [comment]
Welcome to join! Please help, we want PostgreSQL working again :wink:

So you're on PostgreSQL too?

What does your log tell us? This will definitely have something added  to the ownCloud log.

**Comment by richpuch** [comment]
Thanks...

"So you're on PostgreSQL too?"

The answer was no, I was using sqlite3....I converted over to MYSQL and all is fine.

Thanks again

**Comment by fcturner** [comment]
sqlite3 you say. It that any different from sqlite?

**Comment by richpuch** [comment]
Sorry, typo. SQLite.

**Comment by IzK666** [comment]
I don't know if that can help you or not. It didn't work either.... :disappointed: 

I have a machine running Ubuntu 12.04, PHP 5.5.34, Apache 2.4.16, PostgreSQL 9.1.21 and owncloud v9.0.1. I downloaded the latest passwords version from github.

After uncomment some lines in the code to run under PG database, I get a "Decrypting passwords".
It seems that PostGre don't like the query.
db/passswordmapper.php:
Require to define the new data type (missing if (other db) - then - else):
$sql = 'SELECT id::varchar, user_id, loginname, website, address, pass, properties, notes, creation_date, deleted FROM *PREFIX*passwords '

And then, in line 69, can't be used a function:
$sql = $sql . ' ORDER BY website ASC';

With those small changes I get a "No passwords yet" and no errors.
The app let me create new passwords and store in the database, but still don't load the passwords list and shows "No passwords yet".

After that, I added some lines to debug in service/passwordservice.php (lines 22 and 23):
$result = $this->mapper->findAll($userId);
error_log(count($result)); ***// returns 6***
$arr = json_decode(json_encode($result), true);
error_log(count($arr)); ***// returns 0***

The category manager works fine: I can create and delete categories.

Hope it helps somehow.... I don't know where is the error or what to do.
